File: src/GidsMain.java

import javafx.application.Application;
import javafx.geometry.Side;
import javafx.scene.Group;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.*;
import javafx.scene.layout.VBox;
import javafx.stage.Stage;


public class GidsMain extends Application {

    private TabPane tabPane;
    private Tab tab1;
    private Tab tab2;

    public Parent createContent() {
        tabPane = new TabPane();
        tabPane.setPrefSize(600, 500);
        tabPane.setMinSize(TabPane.USE_PREF_SIZE, TabPane.USE_PREF_SIZE);
        tabPane.setMaxSize(TabPane.USE_PREF_SIZE, TabPane.USE_PREF_SIZE);
        tab1 = new Tab();
        tab2 = new Tab();

        tabPane.setRotateGraphic(false);
        tabPane.setTabClosingPolicy(TabPane.TabClosingPolicy.SELECTED_TAB);
        tabPane.setSide(Side.TOP);

        // Drag and Drop tab
        tab1.setText("Drag and Drop");
        tab1.setTooltip(new Tooltip("Drag and Drop example"));
        final Group grp = new Group();  // drag and drop needs a group to setup the pane
        LayoutUI ui = new LayoutUI();
        ui.setUpDragDrop(grp);
        tab1.setContent(grp);
        tabPane.getTabs().add(tab1);

        // Grid clipboard tab
        tab2.setText("Grid Clipboard");
        tab2.setTooltip(new Tooltip("Grid Clipboard example"));
        final VBox vbox = new VBox();
        vbox.setSpacing(10);
        vbox.setTranslateX(10);
        vbox.setTranslateY(10);
        GridUI gui = new GridUI();
        gui.setUpGrid(vbox);
        tab2.setContent(vbox);
        tabPane.getTabs().add(tab2);
      return tabPane;
    }

    @Override public void start(Stage primaryStage) throws Exception {
        primaryStage.setTitle("GIDS JavaFx Demo");
        primaryStage.setScene(new Scene(createContent()));
        primaryStage.show();
    }

    /**
     * Java main for when running without JavaFX launcher
     */
    public static void main(String[] args) {
        launch(args);
    }

}

File: src/main/scala/main/GidsMain.scala
package main

import scalafx.Includes._
import com.westat.{GDSection, GridUI}

import scalafx.application.JFXApp
import scalafx.event.ActionEvent
import scalafx.event.EventIncludes
import scalafx.geometry.{Insets, Side}
import scalafx.scene.Scene
import scalafx.scene.control.TabPane.TabClosingPolicy
import scalafx.scene.control._
import scalafx.scene.layout.{HBox, Priority, VBox}
import scalafx.stage.Screen
/**
 * Created by Owner on 5/3/2017.
 */


object GidsMain extends JFXApp {

  stage = new JFXApp.PrimaryStage {
    //   initStyle(StageStyle.Unified)
    title = "GidsJavaFx"
    width = 1000
    height = 500

    scene = new Scene {
      root = new VBox {
        vgrow = Priority.Always
        hgrow = Priority.Always
        spacing = 10
        padding = Insets(20)
        maxWidth = Screen.primary.visualBounds.width
        maxHeight = Screen.primary.visualBounds.height
        children = List(
          new ToolBar {
            // toolbar controls
            val btnLeft = new Button("Left")
            btnLeft.onAction =  (event : ActionEvent) => {
            //  println("left button clicked")
              GDSection.moveCurrentLeft(edSize.getText)
            }
            val btnRight = new Button("Right")
            btnRight.onAction = (event : ActionEvent) => {
              //println("right button clicked")
              GDSection.moveCurrentRight(edSize.getText)
            }
            val edSize = new TextField()
            edSize.text =  "1.00in"
            edSize.setPromptText("Distance (eg 1.0mm)")

            items = List(
              new Separator(),
              btnLeft,
              btnRight,
              new Separator(),
              edSize,
              new Separator()
            )
          },

        new TabPane {
            minWidth = 400
            prefWidth = 1000
            prefHeight = 500
            minWidth = 500
            minHeight = 300

          tabs = Seq(
              new Tab {
                text = "Grid Clipboard"
                tooltip = new Tooltip("Grid Clipboard example")
                content = new VBox {
                  spacing = 10
                  children = List(
                    new GridUI().tableView
                  )
                }
              },
              new Tab {
                text = "Drag and Drop"
                tooltip = new Tooltip("Drag and Drop example")
                content = new FormDesigner().node
              }

            )
            tabClosingPolicy = TabClosingPolicy.Unavailable
            side = Side.Top
          }

        )
      }
    }
  }


}
